### PR TITLE
temporalite: init at 0.2.0

### PR DESCRIPTION
**Pull request by lucperkins**
###### Description of changes

Adds [Temporalite](https://github.com/temporalio/temporalite), a "light" distribution of the [Temporal](https://temporal.io) server, to Nixpkgs.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by drupol — APPROVED** [review]

**Review by drupol — APPROVED** [review]
LGTM

File: pkgs/applications/networking/cluster/temporalite/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-rLEkWg5LNVb7i/2IARKGuP9ugaVJA9pwYbKLm0QLmOc=";
+  };
+
+  vendorSha256 = "sha256-vjuwh/HRRYfB6J49rfJxif12nYPnbBodWF9hTiGygS8=";
+
+  # The test suite currently involves integration tests that can't be
+  # reproduced in Nix.
+  doCheck = false;
```
Comment (viraptor) [inline pkgs/applications/networking/cluster/temporalite/default.nix:21]: Can we disable just those tests? See https://github.com/NixOS/nixpkgs/blob/2f397f1cca64859ad8897cd66a891e6d6ea3134c/pkgs/tools/text/goawk/default.nix#L19-L23 for an example

**Comment by lucperkins** [comment]
@viraptor Good call. I've done that on two specific tests here. I also add a `subPackages` attribute because it turns out that there's also an executable in `internal/copyright` that I assume we do _not_ want in the final package here.

**Comment by viraptor** [comment]
Sorry, I missed it this time before merging, but please squash the commits next time when you're updating the PR @lucperkins ❤️

**Pull request merged**